Map/RoomBase.gd

```diff
@@ -4,12 +4,18 @@ class_name RoomBase
 ##
 ## Holds basic functionality for a room on the map
 
-@export var light_level: Enums.LightLevel = Enums.LightLevel.UNLIT
+@export var light_level: Enums.LightLevel:
+	set(value):
+		light_level = value
+		on_light_level_changed.emit(light_level)
 @export var has_torch: bool = false
 @export var room_event: EventBase
 
 signal on_light_level_changed
 
+func _init(_light_level: Enums.LightLevel):
```

Comment:
it would be good to have a default argument if we want to create an instance of the class to use it after, without needing to supply an initial state